### PR TITLE
Fix whiteboard problems

### PR DESCRIPTION
**Pull request by ZappoMan**
The problem was that the logic for handling repacking attempts was broken and we would keep attempting to repack over and over again even if things wouldn't fit. This change properly handle repack attempts.

### CODE REVIEW

File: assignment-client/src/octree/OctreeSendThread.cpp [hunk 1]
```diff
@@ -416,27 +416,6 @@ int OctreeSendThread::packetDistributor(OctreeQueryNode* nodeData, bool viewFrus
                         return;
                     }
 
-                    /* TODO: Looking for a way to prevent locking and encoding a tree that is not
```
Comment (ZappoMan) [inline assignment-client/src/octree/OctreeSendThread.cpp]: dead code removal

File: assignment-client/src/octree/OctreeSendThread.cpp [hunk 2]
```diff
@@ -471,20 +450,9 @@ int OctreeSendThread::packetDistributor(OctreeQueryNode* nodeData, bool viewFrus
                     // the packet and send it
                     completedScene = nodeData->elementBag.isEmpty();
 
-                    // if we're trying to fill a full size packet, then we use this logic to determine if we have a DIDNT_FIT case.
```
Comment (ZappoMan) [inline assignment-client/src/octree/OctreeSendThread.cpp]: this really is the bug... there was no reason to special case the "full size packet"... basically if the encode came back as "DIDNT_FIT" then we wanted to drop into the `lastNodeDidntFit` logic. you will also notice the old code wasn't properly incrementing the `extraPackingAttempts`... no in all cases of DIDNT_FIT we consider it a `lastNodeDidntFit` and we track out `extraPackingAttempts`

File: assignment-client/src/octree/OctreeSendThread.cpp [hunk 3]
```diff
@@ -526,8 +493,8 @@ int OctreeSendThread::packetDistributor(OctreeQueryNode* nodeData, bool viewFrus
                 // the packet doesn't have enough space to bother attempting to pack more...
                 bool sendNow = true;
 
-                if (nodeData->getAvailable() >= MINIMUM_ATTEMPT_MORE_PACKING &&
-                    extraPackingAttempts <= REASONABLE_NUMBER_OF_PACKING_ATTEMPTS) {
+                if (!completedScene && (nodeData->getAvailable() >= MINIMUM_ATTEMPT_MORE_PACKING &&
```
Comment (ZappoMan) [inline assignment-client/src/octree/OctreeSendThread.cpp]: In the case of a completed scene we don't want to set the sendNow to false, although this is probably unrelated to the whiteboard problem

File: assignment-client/src/octree/OctreeSendThread.cpp [hunk 4]
```diff
@@ -539,6 +506,7 @@ int OctreeSendThread::packetDistributor(OctreeQueryNode* nodeData, bool viewFrus
                     packetSendingElapsedUsec = (float)(packetSendingEnd - packetSendingStart);
 
                     targetSize = nodeData->getAvailable() - sizeof(OCTREE_PACKET_INTERNAL_SECTION_SIZE);
+                    extraPackingAttempts = 0;
```
Comment (ZappoMan) [inline assignment-client/src/octree/OctreeSendThread.cpp]: also only reset `extraPackingAttempts` if we've actually sent the packet using `handlePacketSend()`

File: assignment-client/src/octree/OctreeSendThread.cpp [hunk 5]
```diff
@@ -517,7 +485,6 @@ int OctreeSendThread::packetDistributor(OctreeQueryNode* nodeData, bool viewFrus
                     }
 
                     nodeData->writeToPacket(_packetData.getFinalizedData(), _packetData.getFinalizedSize());
-                    extraPackingAttempts = 0;
```
Comment (ZappoMan) [inline assignment-client/src/octree/OctreeSendThread.cpp]: this is probably the "money shot" -- basically we kept resetting the `extraPackingAttempts` to `0` and that would mean we kept thinking we had more attempts to repack available... we shouldn't be resetting this until we actually send the packet.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 65e41198bf850752ce850128ccfd896a75aa357e available at: https://gist.github.com/d85eaff9c1ee9dd0b7bb<br/><br/>No installer was built for this commit

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2fbabf20db0879a404f8dafbbdb79c92706bdcab available at: https://gist.github.com/d85eaff9c1ee9dd0b7bb<br/><br/>No installer was built for this commit

**Comment by murillodigital** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 2fbabf20db0879a404f8dafbbdb79c92706bdcab available at: https://gist.github.com/d85eaff9c1ee9dd0b7bb<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/High+Fidelity-PR-6614-win32.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6614.dmg

**Pull request merged**